src/psbt.c

```diff
@@ -2757,6 +2783,16 @@ int wally_finalize_psbt(struct wally_psbt *psbt)
                 if ((ret = wally_witness_p2wpkh_from_der(partial_sig->pubkey, pubkey_len, partial_sig->sig, partial_sig->sig_len, &input->final_witness)) != WALLY_OK) {
                     return ret;
                 }
+                if (input->redeem_script) {
+                    /* P2SH wrapped witness requires final scriptsig of pushing the redeemScript */
+                    script_sig_len = varint_get_length(input->redeem_script_len) + input->redeem_script_len;
+                    input->final_script_sig = wally_malloc(script_sig_len);
```

Comment:
May be worth explicitly checking the value returned by wally_malloc for null and returning WALLY_ENOMEM

Reply:
this is copied from https://github.com/ElementsProject/libwally-core/blob/master/src/psbt.c#L2860-L2868, which will also need to be cleaned up

Reply:
in the interest of getting my PRs in c-lightning unblocked, we should merge this PR and clean up the malloc problems in a second fixup

Reply:
ok, i went through and updated all `wally_alloc` calls in psbt.c that were missing a NULL check, see #204